Use workflow object for rendering sample tags

ReeceHoffmann opened the pull request:
Light refactor of how sample tags are rendered to:
 1. Use the new workflow object
 2. Support aodp workflow tags
 3. Increase flexibility for when new workflows are added
 4. Add workflow typing

ReeceHoffmann committed:
fix: use workflow object for rendering sample tags

igboyes commented on src/js/samples/components/Item/Tags.tsx:
```diff
@@ -0,0 +1,124 @@
+        },
+        [],
+    );
+    return (
```
```suggestion
    );
    
    return (
```

igboyes commented on src/js/samples/components/Item/Tags.tsx:
```diff
@@ -0,0 +1,124 @@
+};
+
+const labels = { pathoscope: "Pathoscope", nuvs: "NuVs", aodp: "AODP" };
+export function SampleItemWorkflowTags({ id, workflows }: SampleItemWorkflowTagsProps) {
```
Look at `getWorkflowDisplayName()`.

```suggestion
const labels = { pathoscope: "Pathoscope", nuvs: "NuVs", aodp: "AODP" };

export function SampleItemWorkflowTags({ id, workflows }: SampleItemWorkflowTagsProps) {
```

igboyes commented on src/js/samples/components/Item/Tags.tsx:
```diff
@@ -0,0 +1,124 @@
+    workflowState: WorkflowState;
+};
+
+export function SampleItemWorkflowTag({ label, workflowState }: SampleItemWorkflowTagProps) {
```
Put in separate module. We should be following one component per module convention.

igboyes reviewed (changes requested):
Please add docstring to exported items.

ReeceHoffmann committed:
chore: address review

  - Separate exported workflow tags into individual files
  - Add jsdoc compliant docstrings for all touched functions

igboyes commented on src/js/samples/components/Tag/Icon.tsx:
```diff
@@ -0,0 +1,11 @@
+/**
+ * Styled component for the icon in a workflow tag.
+ *
+ *@Returns {StyledComponent}
```
```suggestion
 *@returns {StyledComponent}
```

igboyes commented on src/js/samples/components/Tag/Tags.tsx:
```diff
@@ -0,0 +1,55 @@
+        },
+        [],
+    );
+    return (
```
```suggestion
    );
    
    return (
```

igboyes reviewed (changes requested):
F2F discussion summary:
* Fix component naming.
* Fix file naming.
* Use TSDoc spec.
* Document shared types.

ReeceHoffmann committed:
chore: improve documentation

 - Rename workflowTag components
 - Rename workflowTag component modules to match exported component 
 - Add and reword typing for all touched components to use TSdoc styling

igboyes commented on src/js/samples/components/Tag/WorkflowTagLink.tsx:
```diff
@@ -0,0 +1,24 @@
+`;
+
+/**
+ * Display a stylized link to the analyses page for a sample.
```
A link to the analyses page for a sample that is inline with its workflow tags

_or similar_.

igboyes commented on src/js/samples/components/Tag/WorkflowLabelIcon.tsx:
```diff
@@ -0,0 +1,11 @@
+import styled from "styled-components";
+
+/**
+ * Styled component for the icon in a workflow tag.
```
```suggestion
 * The icon in a workflow tag.
```

igboyes commented on src/js/samples/components/Tag/WorkflowTag.tsx:
```diff
@@ -0,0 +1,32 @@
+};
+
+/**
+ * Display a workflow tag
```
What is it exactly?

igboyes replied:
```suggestion
 * A workflow tag.
```

igboyes commented on src/js/samples/components/Tag/WorkflowTagNone.tsx:
```diff
@@ -0,0 +1,27 @@
+`;
+
+/**
+ * Display a "no analyses" tag.
```
Where? When would you want to show this?

```suggestion
 * A  "no analyses" tag.
```

igboyes commented on src/js/samples/components/Tag/WorkflowTagNone.tsx:
```diff
@@ -0,0 +1,27 @@
+/**
+ * Display a "no analyses" tag.
+ *
+ * @returns A tag for a sample item with no analyses.
```
```suggestion
 * @returns A workflow tag for a sample item with no analyses.
```

igboyes commented on src/js/samples/components/Tag/WorkflowTags.tsx:
```diff
@@ -0,0 +1,53 @@
+};
+
+/**
+ * Display workflow tags for a sample
```
```suggestion
 * Workflow tags for a sample item.
 *
 * The tags show what state the sample's analyses are in for each workflow.
```

igboyes commented on src/js/samples/types.ts:
```diff
@@ -0,0 +1,26 @@
+ */
+
+/**
+ * Collection of all workflow states that can be associated with a sample.
```
```suggestion
 * All workflow states that can be associated with a sample.
```

igboyes commented on src/js/samples/types.ts:
```diff
@@ -0,0 +1,26 @@
+}
+
+/**
+ * Collection of all types of workflows and their state associated with a sample
```
```suggestion
 * All workflow states for a sample.
```

igboyes commented on src/js/samples/components/Tag/WorkflowLabelIcon.tsx:
```diff
@@ -0,0 +1,11 @@
+/**
+ * Styled component for the icon in a workflow tag.
+ *
+ * @returns The wrapper class for styling the icon in a workflow tag.
```
This line doesn't make sense to me. I think it should be removed entirely since it is a component.

ReeceHoffmann committed:
docs: adjust docstring phrasing

igboyes reviewed (commented):
Make `Item.jsx` a TSX file and this is ready to go IMO.

ReeceHoffmann committed:
Merge remote-tracking branch 'upstream/main' into reecehoffmann/ui-45-render-tags-based-on-workflows-field

igboyes commented:
Resolve DS issue using unary and I will merge.

ReeceHoffmann committed:
refactor: address ds problem

ghost commented:
:tada: This PR is included in version 3.14.0 :tada:

The release is available on [GitHub release](https://github.com/virtool/virtool-ui/releases/tag/3.14.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
